src/components/atomic-swap/ExternalChangeModal.js

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React, { useEffect } from "react";
+import { t } from "ttag";
+import $ from 'jquery';
+
+export function ModalAtomicExternalChange ({ manageDomLifecycle }) {
+    const modalDomId = 'atomicExternalChangeModal';
+
+    useEffect(() => {
+        manageDomLifecycle(`#${modalDomId}`);
```

Comment:
This `manageDomLifecycle` function was created precisely to avoid accidental memory leaks by centralizing DOM manipulation code.

After testing I found out all listeners were being consistently removed from memory. Here's some evidence from Chromium's `Performance` tab confirming that the cleanup is being successful over time while:
- Opening and dismissing the modals
- Opening the modals and completing their operation ( Send / Receive )
( While these operations aren't precisely from the code you pointed on the comment, they all use the same lifecycle managing method )

![Screenshot from 2023-02-03 19-55-24](https://user-images.githubusercontent.com/1299409/216726620-30229beb-2b04-4a7d-be22-ecab4c8afb66.png)

To be extra sure, however, added a `.off()` call on the hidden modal handler, even though it didn't change the performance pattern noticeably:
![image](https://user-images.githubusercontent.com/1299409/216727863-44b2b1cb-58b5-4095-a0ac-46487e7d52b0.png)